opengrok-indexer/src/main/java/org/opengrok/indexer/index/IndexDatabase.java

```diff
@@ -1629,46 +1666,63 @@ public int hashCode() {
         return hash;
     }
 
-    private boolean isXrefWriter(AbstractAnalyzer fa) {
-        AbstractAnalyzer.Genre g = fa.getFactory().getGenre();
-        return (g == AbstractAnalyzer.Genre.PLAIN || g == AbstractAnalyzer.Genre.XREFABLE);
+    private class CountingWriter extends Writer {
+        private long count;
+        private Writer out;
+
+        CountingWriter(Writer out) {
+            super(out);
+            this.out = out;
+            this.count = 0;
```

Comment:
redundant?

Reply:
Looks like it. I also did bunch of other changes suggester by IDEA.